clientapi/routing/routing.go

```diff
@@ -116,6 +116,12 @@ func Setup(
 				nil, cfg, queryAPI, producer, transactionsCache)
 		}),
 	).Methods(http.MethodPut, http.MethodOptions)
+	r0mux.Handle("/rooms/{roomID}/event/{eventID}",
+		common.MakeAuthAPI("room_get_event", authData, func(req *http.Request, device *authtypes.Device) util.JSONResponse {
+			vars := mux.Vars(req)
```

Comment:
We do this silly thing now for URL encoding purposes:

https://github.com/matrix-org/dendrite/blob/c72517687878267a021641a689fdf73fe6d1c127/clientapi/routing/routing.go#L97